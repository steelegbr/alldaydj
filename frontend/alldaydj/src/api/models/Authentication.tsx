export interface ApiLogin {
    email: string,
    password: string
}

export interface ApiLoginResponse {
    refresh: string,
    access: string
}