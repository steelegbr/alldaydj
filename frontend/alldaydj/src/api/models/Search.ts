export interface CartSearchResult {
    label: string,
    id: string,
    title: string,
    artist: string,
    year: number
}

export interface CartSearchResults {
    count: number,
    next?: string,
    previous?: string,
    results: CartSearchResult[]
}

export type CartSearchConditionFields = 'advanced' | 'search' | 'artist' | 'title' | 'page' | 'resultsPerPage';

export type CartSearchConditions = Record<CartSearchConditionFields, string>;
