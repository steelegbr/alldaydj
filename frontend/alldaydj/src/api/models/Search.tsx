export interface CartSearchResult {
    label: string,
    id: string,
    title: string,
    artist: string,
    year: number
}

export interface CartSearchResults {
    count: number,
    next?: string,
    previous?: string,
    results: CartSearchResult[]
}

export type SearchConditionFields = 'advanced' | 'search' | 'artist' | 'title';

export type SearchConditions = Record<SearchConditionFields, string>;
