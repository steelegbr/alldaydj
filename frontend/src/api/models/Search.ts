/**
    AllDay DJ - Radio Automation
    Copyright (C) 2020-2021 Marc Steele

    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <https://www.gnu.org/licenses/>.
*/

export interface CartSearchResult {
    label: string,
    id: string,
    title: string,
    artist: string,
    year: number
}

export interface CartSearchResults {
    count: number,
    next?: string,
    previous?: string,
    results: CartSearchResult[]
}

export type CartSearchConditionFields = 'search' | 'page' | 'resultsPerPage';

export type CartSearchConditions = Record<CartSearchConditionFields, string>;
