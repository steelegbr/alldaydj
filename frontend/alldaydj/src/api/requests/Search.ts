/* eslint-disable import/prefer-default-export */
import axios, { AxiosResponse } from 'axios';
import getUrl from 'services/UrlService';
import { CartSearchResults, CartSearchConditions } from 'api/models/Search';
import { generateRequestConfig } from 'api/requests/Helpers';

const wildcardWrap = (term: string) : string => (term ? `*${term}*` : '*');

export const cartSearch = (
  conditions: CartSearchConditions,
  token: string,
): Promise<AxiosResponse<CartSearchResults>> => {
  const params = new URLSearchParams();
  params.append('search', wildcardWrap(conditions.search));
  params.append('page', conditions.page);
  params.append('page_size', conditions.resultsPerPage);

  if (conditions.advanced === 'true') {
    params.append('search', `title:${wildcardWrap(conditions.title)}`);
    params.append('search', `artist:${wildcardWrap(conditions.artist)}`);
  }

  return axios.get<CartSearchResults>(getUrl('/api/cart/search/'), generateRequestConfig(token, params));
};
