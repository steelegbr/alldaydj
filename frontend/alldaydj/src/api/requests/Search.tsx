/* eslint-disable import/prefer-default-export */
import axios, { AxiosResponse } from 'axios';
import getUrl from '../../services/UrlService';
import { CartSearchResults, SearchConditions } from '../models/Search';
import { generateRequestConfig } from './Helpers';

interface CartSearchParams {
    search: string,
    page: number,
    // eslint-disable-next-line camelcase
    page_size: number,
    title?: string,
    artist?: string,
    year?: number
}

const wildcardWrap = (term: string) : string => `*${term}*`;

export const cartSearch = (
  conditions: SearchConditions,
  page: number,
  pageSize: number,
  token: string,
): Promise<AxiosResponse<CartSearchResults>> => {
  const params : CartSearchParams = {
    search: wildcardWrap(conditions.search),
    page,
    page_size: pageSize,
  };

  if (conditions.advanced === 'true') {
    params.title = wildcardWrap(conditions.title);
    params.artist = wildcardWrap(conditions.artist);
  }

  return axios.get<CartSearchResults>(getUrl('/api/cart/search/'), generateRequestConfig(token, params));
};
