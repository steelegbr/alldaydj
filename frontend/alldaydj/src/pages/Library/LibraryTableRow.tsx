import {
  Button, Menu, MenuItem, TableCell, TableRow,
} from '@material-ui/core';
import { GetApp, PlayArrow } from '@material-ui/icons';
import React from 'react';
import { CartSearchResult } from '../../api/models/Search';

interface TableRowProps {
    result: CartSearchResult
}

const LibraryTableRow = ({ result }: TableRowProps) => {
  const [anchorButton, setAnchorButton] = React.useState<HTMLElement | undefined>(undefined);

  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
    setAnchorButton(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorButton(undefined);
  };

  return (
    <TableRow>
      <TableCell>{result.label}</TableCell>
      <TableCell>{result.artist}</TableCell>
      <TableCell>{result.title}</TableCell>
      <TableCell>
        <Button aria-controls="download audio" aria-haspopup="true" onClick={handleClick}>
          <GetApp />
        </Button>
        <Menu
          anchorEl={anchorButton}
          id="menu-audio-download"
          keepMounted
          onClose={handleClose}
          open={Boolean(anchorButton)}
        >
          <MenuItem onClick={handleClose}>Compressed (OGG)</MenuItem>
          <MenuItem onClick={handleClose}>Uncompressed (WAV)</MenuItem>
        </Menu>
      </TableCell>
      <TableCell>
        <Button>
          <PlayArrow />
        </Button>
      </TableCell>
    </TableRow>
  );
};

export default LibraryTableRow;
