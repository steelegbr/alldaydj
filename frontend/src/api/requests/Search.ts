/* eslint-disable import/prefer-default-export */
import axios, { AxiosResponse } from 'axios';
import getUrl from 'services/UrlService';
import { CartSearchConditions, CartSearchResult } from 'api/models/Search';
import { generateRequestConfig } from 'api/requests/Helpers';
import { Paginated } from 'api/models/Pagination';

export const cartSearch = (
  conditions: CartSearchConditions,
): Promise<AxiosResponse<Paginated<CartSearchResult>>> => {
  const params = new URLSearchParams();
  params.append('search', conditions.search);
  params.append('page', conditions.page);
  params.append('page_size', conditions.resultsPerPage);
  return axios.get<Paginated<CartSearchResult>>(getUrl('/api/cart/search/'), generateRequestConfig(params));
};
