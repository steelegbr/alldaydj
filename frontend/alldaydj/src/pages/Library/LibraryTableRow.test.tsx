import React from 'react';
import { mount } from 'enzyme';
import { CartSearchResult } from '../../api/models/Search';
import LibraryTableRow from './LibraryTableRow';

const sampleResult : CartSearchResult = {
  id: '957dbe30-007d-442e-975e-42e096e60fa2',
  label: 'abc123',
  title: 'Cart Title',
  artist: 'Artist Name',
  year: 1988,
};

describe('library table row', () => {
  const getComponent = (result: CartSearchResult) => mount(<LibraryTableRow result={result} />);

  it('renders single result', () => {
    const component = getComponent(sampleResult);
    expect(component).toMatchSnapshot();
  });
});
