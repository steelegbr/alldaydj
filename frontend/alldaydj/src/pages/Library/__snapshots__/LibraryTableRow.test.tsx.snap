// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`library table row renders single result 1`] = `
<LibraryTableRow
  result={
    Object {
      "artist": "Artist Name",
      "id": "957dbe30-007d-442e-975e-42e096e60fa2",
      "label": "abc123",
      "title": "Cart Title",
      "year": 1988,
    }
  }
>
  <WithStyles(ForwardRef(TableRow))>
    <ForwardRef(TableRow)
      classes={
        Object {
          "footer": "MuiTableRow-footer",
          "head": "MuiTableRow-head",
          "hover": "MuiTableRow-hover",
          "root": "MuiTableRow-root",
          "selected": "Mui-selected",
        }
      }
    >
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <WithStyles(ForwardRef(TableCell))>
          <ForwardRef(TableCell)
            classes={
              Object {
                "alignCenter": "MuiTableCell-alignCenter",
                "alignJustify": "MuiTableCell-alignJustify",
                "alignLeft": "MuiTableCell-alignLeft",
                "alignRight": "MuiTableCell-alignRight",
                "body": "MuiTableCell-body",
                "footer": "MuiTableCell-footer",
                "head": "MuiTableCell-head",
                "paddingCheckbox": "MuiTableCell-paddingCheckbox",
                "paddingNone": "MuiTableCell-paddingNone",
                "root": "MuiTableCell-root",
                "sizeSmall": "MuiTableCell-sizeSmall",
                "stickyHeader": "MuiTableCell-stickyHeader",
              }
            }
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root"
            >
              <WithStyles(ForwardRef(IconButton))
                aria-label="expand search result"
                onClick={[Function]}
                size="small"
              >
                <ForwardRef(IconButton)
                  aria-label="expand search result"
                  classes={
                    Object {
                      "colorInherit": "MuiIconButton-colorInherit",
                      "colorPrimary": "MuiIconButton-colorPrimary",
                      "colorSecondary": "MuiIconButton-colorSecondary",
                      "disabled": "Mui-disabled",
                      "edgeEnd": "MuiIconButton-edgeEnd",
                      "edgeStart": "MuiIconButton-edgeStart",
                      "label": "MuiIconButton-label",
                      "root": "MuiIconButton-root",
                      "sizeSmall": "MuiIconButton-sizeSmall",
                    }
                  }
                  onClick={[Function]}
                  size="small"
                >
                  <WithStyles(ForwardRef(ButtonBase))
                    aria-label="expand search result"
                    centerRipple={true}
                    className="MuiIconButton-root MuiIconButton-sizeSmall"
                    disabled={false}
                    focusRipple={true}
                    onClick={[Function]}
                  >
                    <ForwardRef(ButtonBase)
                      aria-label="expand search result"
                      centerRipple={true}
                      className="MuiIconButton-root MuiIconButton-sizeSmall"
                      classes={
                        Object {
                          "disabled": "Mui-disabled",
                          "focusVisible": "Mui-focusVisible",
                          "root": "MuiButtonBase-root",
                        }
                      }
                      disabled={false}
                      focusRipple={true}
                      onClick={[Function]}
                    >
                      <button
                        aria-label="expand search result"
                        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
                        disabled={false}
                        onBlur={[Function]}
                        onClick={[Function]}
                        onDragLeave={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseDown={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchMove={[Function]}
                        onTouchStart={[Function]}
                        tabIndex={0}
                        type="button"
                      >
                        <span
                          className="MuiIconButton-label"
                        >
                          <ForwardRef(KeyboardArrowDownIcon)>
                            <WithStyles(ForwardRef(SvgIcon))>
                              <ForwardRef(SvgIcon)
                                classes={
                                  Object {
                                    "colorAction": "MuiSvgIcon-colorAction",
                                    "colorDisabled": "MuiSvgIcon-colorDisabled",
                                    "colorError": "MuiSvgIcon-colorError",
                                    "colorPrimary": "MuiSvgIcon-colorPrimary",
                                    "colorSecondary": "MuiSvgIcon-colorSecondary",
                                    "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                                    "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                                    "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                                    "root": "MuiSvgIcon-root",
                                  }
                                }
                              >
                                <svg
                                  aria-hidden={true}
                                  className="MuiSvgIcon-root"
                                  focusable="false"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                                  />
                                </svg>
                              </ForwardRef(SvgIcon)>
                            </WithStyles(ForwardRef(SvgIcon))>
                          </ForwardRef(KeyboardArrowDownIcon)>
                        </span>
                        <WithStyles(memo)
                          center={true}
                        >
                          <ForwardRef(TouchRipple)
                            center={true}
                            classes={
                              Object {
                                "child": "MuiTouchRipple-child",
                                "childLeaving": "MuiTouchRipple-childLeaving",
                                "childPulsate": "MuiTouchRipple-childPulsate",
                                "ripple": "MuiTouchRipple-ripple",
                                "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                                "rippleVisible": "MuiTouchRipple-rippleVisible",
                                "root": "MuiTouchRipple-root",
                              }
                            }
                          >
                            <span
                              className="MuiTouchRipple-root"
                            >
                              <TransitionGroup
                                childFactory={[Function]}
                                component={null}
                                exit={true}
                              />
                            </span>
                          </ForwardRef(TouchRipple)>
                        </WithStyles(memo)>
                      </button>
                    </ForwardRef(ButtonBase)>
                  </WithStyles(ForwardRef(ButtonBase))>
                </ForwardRef(IconButton)>
              </WithStyles(ForwardRef(IconButton))>
            </td>
          </ForwardRef(TableCell)>
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))>
          <ForwardRef(TableCell)
            classes={
              Object {
                "alignCenter": "MuiTableCell-alignCenter",
                "alignJustify": "MuiTableCell-alignJustify",
                "alignLeft": "MuiTableCell-alignLeft",
                "alignRight": "MuiTableCell-alignRight",
                "body": "MuiTableCell-body",
                "footer": "MuiTableCell-footer",
                "head": "MuiTableCell-head",
                "paddingCheckbox": "MuiTableCell-paddingCheckbox",
                "paddingNone": "MuiTableCell-paddingNone",
                "root": "MuiTableCell-root",
                "sizeSmall": "MuiTableCell-sizeSmall",
                "stickyHeader": "MuiTableCell-stickyHeader",
              }
            }
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root"
            >
              abc123
            </td>
          </ForwardRef(TableCell)>
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))>
          <ForwardRef(TableCell)
            classes={
              Object {
                "alignCenter": "MuiTableCell-alignCenter",
                "alignJustify": "MuiTableCell-alignJustify",
                "alignLeft": "MuiTableCell-alignLeft",
                "alignRight": "MuiTableCell-alignRight",
                "body": "MuiTableCell-body",
                "footer": "MuiTableCell-footer",
                "head": "MuiTableCell-head",
                "paddingCheckbox": "MuiTableCell-paddingCheckbox",
                "paddingNone": "MuiTableCell-paddingNone",
                "root": "MuiTableCell-root",
                "sizeSmall": "MuiTableCell-sizeSmall",
                "stickyHeader": "MuiTableCell-stickyHeader",
              }
            }
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root"
            >
              Artist Name
            </td>
          </ForwardRef(TableCell)>
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))>
          <ForwardRef(TableCell)
            classes={
              Object {
                "alignCenter": "MuiTableCell-alignCenter",
                "alignJustify": "MuiTableCell-alignJustify",
                "alignLeft": "MuiTableCell-alignLeft",
                "alignRight": "MuiTableCell-alignRight",
                "body": "MuiTableCell-body",
                "footer": "MuiTableCell-footer",
                "head": "MuiTableCell-head",
                "paddingCheckbox": "MuiTableCell-paddingCheckbox",
                "paddingNone": "MuiTableCell-paddingNone",
                "root": "MuiTableCell-root",
                "sizeSmall": "MuiTableCell-sizeSmall",
                "stickyHeader": "MuiTableCell-stickyHeader",
              }
            }
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root"
            >
              Cart Title
            </td>
          </ForwardRef(TableCell)>
        </WithStyles(ForwardRef(TableCell))>
      </tr>
    </ForwardRef(TableRow)>
  </WithStyles(ForwardRef(TableRow))>
  <WithStyles(ForwardRef(TableRow))>
    <ForwardRef(TableRow)
      classes={
        Object {
          "footer": "MuiTableRow-footer",
          "head": "MuiTableRow-head",
          "hover": "MuiTableRow-hover",
          "root": "MuiTableRow-root",
          "selected": "Mui-selected",
        }
      }
    >
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <WithStyles(ForwardRef(TableCell))
          className="makeStyles-collapsedRow-1"
          colSpan={4}
        >
          <ForwardRef(TableCell)
            className="makeStyles-collapsedRow-1"
            classes={
              Object {
                "alignCenter": "MuiTableCell-alignCenter",
                "alignJustify": "MuiTableCell-alignJustify",
                "alignLeft": "MuiTableCell-alignLeft",
                "alignRight": "MuiTableCell-alignRight",
                "body": "MuiTableCell-body",
                "footer": "MuiTableCell-footer",
                "head": "MuiTableCell-head",
                "paddingCheckbox": "MuiTableCell-paddingCheckbox",
                "paddingNone": "MuiTableCell-paddingNone",
                "root": "MuiTableCell-root",
                "sizeSmall": "MuiTableCell-sizeSmall",
                "stickyHeader": "MuiTableCell-stickyHeader",
              }
            }
            colSpan={4}
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root makeStyles-collapsedRow-1"
              colSpan={4}
            >
              <WithStyles(ForwardRef(Collapse))
                in={false}
                timeout="auto"
                unmountOnExit={true}
              >
                <ForwardRef(Collapse)
                  classes={
                    Object {
                      "container": "MuiCollapse-container",
                      "entered": "MuiCollapse-entered",
                      "hidden": "MuiCollapse-hidden",
                      "wrapper": "MuiCollapse-wrapper",
                      "wrapperInner": "MuiCollapse-wrapperInner",
                    }
                  }
                  in={false}
                  timeout="auto"
                  unmountOnExit={true}
                >
                  <Transition
                    addEndListener={[Function]}
                    appear={false}
                    enter={true}
                    exit={true}
                    in={false}
                    mountOnEnter={false}
                    onEnter={[Function]}
                    onEntered={[Function]}
                    onEntering={[Function]}
                    onExit={[Function]}
                    onExited={[Function]}
                    onExiting={[Function]}
                    timeout={null}
                    unmountOnExit={true}
                  />
                </ForwardRef(Collapse)>
              </WithStyles(ForwardRef(Collapse))>
            </td>
          </ForwardRef(TableCell)>
        </WithStyles(ForwardRef(TableCell))>
      </tr>
    </ForwardRef(TableRow)>
  </WithStyles(ForwardRef(TableRow))>
</LibraryTableRow>
`;
